src/client/lobby/room-creator.js

```diff
@@ -4,34 +4,39 @@ import config from '../config/config';
 
 import style from './lobby.scss';
 
-export default class RoomCreator extends Component {
-
-    onUsernameClick(){
-        var username = document.getElementById('user-name').value;
-        this.props.setUsername(username);
-    }
+var username;
+var roomname;
 
-    onRoomnameClick(){
-        var roomname = document.getElementById('room-name').value;
-        this.props.setRoomname(roomname);
+export default class RoomCreator extends Component {
+    componentDidMount(){
+        username = document.getElementById('user-name');
+        roomname = document.getElementById('room-name');
     }
-
     render() {
         return (
             <div className="col-lg-4 text-center">
                 <h2>Create Room & Set Username</h2>
                 <div className="col-lg-12 list-group">
                     <div className={`row ${style.space}`}>
                         <input id="user-name" type="text" placeholder="Please type your new username"></input>
-                        <button id="user-name-button" className="btn btn-success" onClick={this.onUsernameClick.bind(this)}>Set Username</button>
+                        <button id="user-name-button" className="btn btn-success" onClick={onUsernameClick.bind(this)}>Set Username</button>
                     </div>
                     <br></br>
                     <div className="row">
                         <input id="room-name" type="text" placeholder="Please type new room name"></input>
-                        <button id="room-name-button" className="btn btn-success" onClick={this.onRoomnameClick.bind(this)}>Create Room</button>
+                        <button id="room-name-button" className="btn btn-success" onClick={onRoomnameClick.bind(this)}>Create Room</button>
                     </div>
                 </div>
             </div>
         );
     }
 };
+
+function onUsernameClick() {
+    cloak.message('setUsername', username.value);
```

Comment:
cloak is a global variable, which is just how the framework works, however, we can improve the situation by wrapping all calls to cloak in a CloakService (this can go with the RoutingService in the services folder)